plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

```diff
@@ -628,9 +628,19 @@ jQuery( function ( $ ) {
 		}
 	);
 
-	$( '.product_attributes' ).on( 'click', '.remove_row', function () {
-		if ( window.confirm( woocommerce_admin_meta_boxes.remove_attribute ) ) {
-			var $parent = $( this ).parent().parent();
+	$( '#product_attributes' ).on( 'click', '.product_attributes .remove_row', function () {
+		var confirmMessage = woocommerce_admin_meta_boxes.remove_attribute;
+		var $parent = $( this ).parent().parent();
+		if ( !$parent.is( '.taxonomy' ) ) {
```

Comment:
I think that this `if` will prevent showing the message when we delete a global attribute (even when it's actually being used for variations).
Maybe we should only check whether the 'Used for variations' checkbox is selected.